designer/public/js/index.js

```diff
@@ -354,6 +355,226 @@ $(function () {
+        brokers =[]
+        var discoverReq = {}
+        
+        discoverReq.entities = [{ type: 'IoTBroker', isPattern: true }];
+
+        var ngsi9client = new NGSI9Client(config.discoveryURL)
+        console.log('discorvery url is ',config.discoveryURL);
+        displayEdgeNode();
+
+        ngsi9client.discoverContextAvailability(discoverReq).then(function (response) {
+            console.log("broker response: ",response);
+            
+            if (response.errorCode.code == 200 && response.hasOwnProperty('contextRegistrationResponses')) {
+                for (var i in response.contextRegistrationResponses) {
+                    var tmpIn = i;
+                    var contextRegistrationResponse = response.contextRegistrationResponses[i];
+                    console.log("broker details is *** ",contextRegistrationResponse);
+                    var brokerID = contextRegistrationResponse.contextRegistration.entities[0].id;
+                    var providerURL = contextRegistrationResponse.contextRegistration.providingApplication;
+                    if (providerURL != '') {
+                        brokers.push({ id: brokerID, brokerURL: providerURL });
+                        console.log("inside broker ",brokers);
+                        if (i==0) {
+                            var option = document.createElement("option");
+                            option.text ='All'; 
+                            var allEdgeBrokerList = document.getElementById("allEdgeBrokerList");
+                            allEdgeBrokerList.add(option);
+                        }
+                        var option = document.createElement("option");
+                        option.text =brokerID.replace('Broker','Edge'); 
+                        console.log("get broker ID: " + brokerID);
+                        var allEdgeBrokerList = document.getElementById("allEdgeBrokerList");
+                        console.log("all edge obje ",allEdgeBrokerList);
+                        allEdgeBrokerList.add(option);
+                        if (tmpIn+1==response.contextRegistrationResponses.length){
+                            if($('#allEdgeBrokerList option:selected').val() == 'All'){
+                                getWorkerList(displayEdgeOnMap,brokers)
+                            }
+                        }
+                    }
+                }
+                
+            }
+
+        }).catch(function (error) {
+            console.log(error);
+            console.log('failed to query context');
+        });
+       // return brokers;
+    }
+    var workerWithDeviceList = {};
+    function getWorkerList(callback,brokerObj) {
+        console.log("inside in get worker list ",brokerObj);
+        var edgeBrokerEntityList = [];
+        workerWithDeviceList = {};
+        for (var i = 0; i < brokerObj.length; i++){
+            var tmpI = i;
+            var tmpClient = new NGSI10Client(brokerObj[i].brokerURL);
```

Comment:
NGSILDclient should be called for getting the entities information ,instead of NGSI10client and whole implementation should be changed accordingly.